torchaudio/functional.py

```diff
@@ -151,7 +152,8 @@ def istft(
     assert 0 < win_length <= n_fft
 
     if window is None:
-        window = torch.ones(win_length, requires_grad=False, device=device, dtype=dtype)
+       window = torch.ones(win_length)
```

Comment:
Is there a reason to create the tensor and then move to a different device?

Reply:
It has to do with the JITability of the function: the signature of `torch.ones` seem not to allow for `device` and `dtype` setting when the `out` parameter is not specified.

Reply:
The `ones` and `eye` tensor can be created on device directly. No need for two separate steps.

Reply:
The last time I checked, the function signatures required an `out` parameter when specifying the device. Since `out` is of type `Tensor` and `window` is of `Optional[Tensor]`, it complained. I might be misinterpreting the errors, though.